cpp/src/arrow/ipc/feather.cc

```diff
@@ -659,6 +670,32 @@ class TableWriter::TableWriterImpl : public ArrayVisitor {
     return Status::OK();
   }
 
+  template <typename ArrayType>
+  Status WriteBinaryArray(const ArrayType& values, ArrayMetadata* meta,
+                          const uint8_t** values_buffer, int64_t* values_bytes,
+                          int64_t* bytes_written) {
+    using offset_type = typename ArrayType::offset_type;
+
+    int64_t offset_bytes = sizeof(offset_type) * (values.length() + 1);
```

Comment:
Would it make sense to push this method to ArrayType?

Reply:
I fail to find a good, non-confusing name for it... Any idea?

Reply:
CalculateOffsetBufferLength?

Reply:
The offsets buffer could be larger, e.g. if the array is sliced.